rotkehlchen/chain/ethereum/modules/curve/savings/decoder.py

```diff
@@ -0,0 +1,88 @@
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == assets_amount and
+                    event.address == SCRV_TOKEN_ADDRESS
+            ):
+                event.counterparty = CPT_CURVE
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                event.notes = f'Deposit {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} in Curve Savings'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == shares_amount and
+                    event.address == ZERO_ADDRESS
+            ):
+                event.counterparty = CPT_CURVE
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {shares_amount} {event.asset.resolve_to_asset_with_symbol().symbol} from depositing into Curve Savings'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == shares_amount and
+                    event.address == ZERO_ADDRESS
+            ):
+                event.counterparty = CPT_CURVE
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {shares_amount} {event.asset.resolve_to_asset_with_symbol().symbol} into Curve Savings'  # noqa: E501
+
+            elif (
+                    event.event_type == HistoryEventType.RECEIVE and
+                    event.event_subtype == HistoryEventSubType.NONE and
+                    event.amount == assets_amount and
+                    event.address == SCRV_TOKEN_ADDRESS
+            ):
+                event.counterparty = CPT_CURVE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Remove {assets_amount} {event.asset.resolve_to_asset_with_symbol().symbol} from Curve Savings'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {SCRV_TOKEN_ADDRESS: (self._decode_deposit_or_withdrawal,)}
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (CounterpartyDetails(identifier=CPT_CURVE, label='Curve.fi', image='curve.png'),)
```

Comment:
maybe it makes sense to store this as a constant and reuse it in the protocols. To not have to update it in all places and avoid creating several times the same object